[material-ui][docs] Add aria-current for nav tabs demo

Kimzify opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

This PR changed `aria-selected` to `aria-current` in `Tab` component. 

From the MDN docs:

> When you have a group of related elements, such as several links in a breadcrumb or steps in a multi-step flow, with one element in the group styled differently from the others to indicate to the sighted user that this is the current element within its group, the aria-current should be used to inform the assistive technology user what has been indicated via styling.


the aria-selected docs define the following.

> The [option](https://w3c.github.io/aria/#option), [tab](https://w3c.github.io/aria/#tab), and [treeitem](https://w3c.github.io/aria/#treeitem) roles permit user agents to provide an implicit value for [aria-selected](https://w3c.github.io/aria/#aria-selected) when specified conditions are met

Closes #32924 

https://deploy-preview-39594--material-ui.netlify.app/material-ui/react-tabs/#nav-tabs

michaldudak commented:
[ARIA docs](https://w3c.github.io/aria/#tab) explicitly state that the active tab should have the `aria-selected` attribute set to `true`:

> Authors SHOULD ensure that a selected tab has its [aria-selected](https://w3c.github.io/aria/#aria-selected) attribute set to true, that inactive tab elements have their [aria-selected](https://w3c.github.io/aria/#aria-selected) attribute set to false, and that the currently selected tab provides a visual indication that it is selected.

Kimzify commented:
> [ARIA docs](https://w3c.github.io/aria/#tab) explicitly state that the active tab should have the `aria-selected` attribute set to `true`:
> 
> > Authors SHOULD ensure that a selected tab has its [aria-selected](https://w3c.github.io/aria/#aria-selected) attribute set to true, that inactive tab elements have their [aria-selected](https://w3c.github.io/aria/#aria-selected) attribute set to false, and that the currently selected tab provides a visual indication that it is selected.

I've updated the PR and add `aria-current` as a separate props to tab component. Also as `aria-current` indicates visually styled  I've changed style's conditions to be chosen by `aria-current` and I've added some related tests as well. Could you please check it and tell me if there is any issues?

michaldudak commented:
Why would we need both? Neither ARIA docs nor demos for the tab role mention the `aria-current` attribute. Is there an accessibility issue you're trying to fix?

Kimzify commented:
> Why would we need both? Neither ARIA docs nor demos for the tab role mention the `aria-current` attribute. Is there an accessibility issue you're trying to fix?

there was an open issue for that and I have trying to fix #32924 
am I getting the issue wrong?

michaldudak commented:
Sorry, I should have explained this better. The issue is only about the navigation tabs demo (https://mui.com/material-ui/react-tabs/#nav-tabs), not the tabs in general.

Kimzify commented:
> Sorry, I should have explained this better. I haven't noticed at first that you're fixing #32924. The issue is only about the navigation tabs demo (https://mui.com/material-ui/react-tabs/#nav-tabs), not the tabs in general.

thank you so much. Just I have another question to clarify solution. 
1. we need `Tab` component to detect links by itself(with `component="a"` props) and add `aria-current` label instead of `aria-selected`
or
2. just we need to update document for Nav tabs to contain aria-current beside aria-selected?

Kimzify commented:
As I was checked, if `aria-current` and `aria-selected` both have the same value, `aria-selected` is preferred to be used. If tabs represent navigations, `aria-current="page"` should be used to indicate navigation usage. 
[from w3 doc:](https://www.w3.org/TR/wai-aria-1.1/#aria-current)

> In some use cases for widgets that support aria-selected, current and selected can have different meanings and can both be used within the same set of elements. For example, aria-current="page" can be used in a navigation [tree](https://www.w3.org/TR/wai-aria-1.1/#tree) to indicate which page is currently displayed, while aria-selected="true" indicates which page will be displayed if the user activates the [treeitem](https://www.w3.org/TR/wai-aria-1.1/#treeitem)

@michaldudak, could you please check the PR again?

siriwatknp commented on docs/data/material/components/tabs/NavTabs.tsx:
```diff
@@ -34,6 +35,7 @@ function LinkTab(props: LinkTabProps) {
           event.preventDefault();
         }
       }}
+      aria-current={props.selected && "page"}
```
```suggestion
      aria-current={props.selected ? "page" : undefined}
```

siriwatknp replied:
Also, please add `aria-selected={undefined}` at the end to override the default behavior of the Tab.

```js
  <Tab
      component="a"
      onClick={(event) => {
        // Routing libraries handle this, you can remove the onClick handle when using them.
        if (samePageLinkNavigation(event)) {
          event.preventDefault();
        }
      }}
      aria-current={props.selected && "page"}
      {...props}
      aria-selected={undefined}
    />
```

Kimzify replied:
as I was checked, based on [w3 document](https://www.w3.org/TR/wai-aria-1.1/#aria-current) the default value of `aria-current` is false. 

> If the attribute is not present or its value is an empty string or undefined, the default value of false applies and the aria-current [state](https://www.w3.org/TR/wai-aria-1.1/#dfn-state) MUST NOT be exposed by user agents or assistive technologies.

Kimzify replied:
and for the second issue, we still need `aria-selected ` tag, because `aria-selected ` and `aria-current` represents different concepts and meanings here. 

based on [w3 document](https://www.w3.org/TR/wai-aria-1.1/#aria-current): 

> In some use cases for widgets that support aria-selected, current and selected can have different meanings and can both be used within the same set of elements.

siriwatknp replied:
Thanks for the info!

siriwatknp reviewed (changes requested):
Based on https://github.com/mui/material-ui/issues/32924#issuecomment-1652251490, the Tabs should have `role="navigation"` as well.

When you update the code, please run `yarn docs:typescript:formatted`.

Kimzify commented:
> Based on [#32924 (comment)](https://github.com/mui/material-ui/issues/32924#issuecomment-1652251490), the Tabs should have `role="navigation"` as well.
> 
> When you update the code, please run `yarn docs:typescript:formatted`.

Done!

siriwatknp commented on docs/data/material/components/tabs/NavTabs.js:
```diff
@@ -27,11 +28,17 @@ function LinkTab(props) {
         }
       }}
+      aria-current={props.selected && 'page'}
+      role="navigation"
```
```suggestion
```

`role="navigation"` should be at Tabs element, not Tab.

Kimzify replied:
Fixed.

Kimzify commented:
@siriwatknp @michaldudak, current PR is ready to review again. could you please check it if you have time?

siriwatknp reviewed (approved):
👍 Thanks for the improvements!
